### PR TITLE
Do not cut git describe in custom release builds

### PR DESCRIPTION
**Pull request by grische**
Currently if autoupdate is enabled on a custom next build, it will
automatically downgrade to the latest release. This change will use
the full git-describe name to avoid those downgrades.

Exemplatory tag name:
One commit on top of latest release "v2021.8.1"

Before: `v2021.8.1~exp2021083100`
After:  `v2021.8.1-1-g2f13e5e~exp20210831`

### CODE REVIEW

**Review by krombel — COMMENTED** [review]
"backport" of #145 but as next will be merged into stable in future either way I would not merge it in general (as it will cause merge conflicts later on)

**Review by goligo — APPROVED** [review]

**Review by awlx — CHANGES_REQUESTED** [review]
Could you reopen the PR with branch next as base? Otherwise as soon as we merge next with stable this will cause a conflict.

**Comment by grische** [comment]
@awlx this has been merged to next as part of #145 already. I opened this PR as the commit in #145 will not apply cleanly because next used `...~next...`  before while stable always had `...~exp...` in the release name.

**Comment by goligo** [comment]
Honestly, I don't see an issue with merging, there is hardly any code in site-ffm.

**Comment by awlx** [comment]
Why can't we just cherry-pick that from next? It's the same and because of the same ref won't cause issues in the future.

**Review by awlx — APPROVED** [review]

**Pull request merged**
